src/track/serato/tags.cpp

```diff
@@ -171,9 +178,22 @@ QList<CueInfo> SeratoTags::getCues(const QString& filePath) const {
             qWarning() << "SeratoTags::getCues: Cue with number < 0 found!";
         }
 
-        if (cueInfo.getType() != CueType::HotCue) {
-            qWarning() << "SeratoTags::getCues: Ignoring cue with non-hotcue type!";
-            continue;
+        switch (cueInfo.getType()) {
+        case CueType::HotCue: {
+            if (index >= kLoopIndexOffset) {
+                qWarning()
+                        << "SeratoTags::getCues: Non-loop Cue with number >="
+                        << kLoopIndexOffset << "found!";
+                continue;
+            }
+            break;
+        }
+        case CueType::Loop: {
+            index += kLoopIndexOffset;
+            break;
+        }
+        default:
+            break;
```

Comment:
This code is repeated between the two loops. Please factor it out to a common function.